src/components/AssetSearch/AssetSearch.tsx

```diff
@@ -160,6 +162,12 @@ export const AssetSearch: FC<AssetSearchProps> = ({
   }, [searchString, searching, sortedAssets])
 
   const listAssets = searching ? searchTermAssets : sortedAssets
+
+  const filteredChainIdsByMarketCap: ChainId[] = useMemo(
+    () => intersection(chainIdsByMarketCap, uniq(listAssets.map(a => a.chainId))),
```

Comment:
Nitpick, `uniqBy` would work with implicit iteration, but I totally get it if we don't want to use unsafe `foo.bar.baz` DSL

Reply:
ya lodash DSL is just pain waiting to happen, big `1efuck` vibes